pushy/src/test/java/com/turo/pushy/apns/ApnsChannelPoolTest.java

```diff
@@ -214,4 +214,34 @@ public void testPendingAcquisitionsDuringPoolClosure() throws Exception {
         assertTrue(pendingFuture.await(1000));
         assertFalse(pendingFuture.isSuccess());
     }
+
+    @Test
+    public void testPendingCreateChannelFutureDuringPoolClosure() throws Exception {
+        final Promise<Void> createSuccess = new DefaultPromise<>(EVENT_EXECUTOR);
+        final PooledObjectFactory<Channel> factory = new PooledObjectFactory<Channel>() {
+            @Override
+            public Future<Channel> create(final Promise<Channel> promise) {
+                EVENT_EXECUTOR.schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        promise.trySuccess(new TestChannel(true));
+                        createSuccess.setSuccess(null);
+                    }
+                }, 1, TimeUnit.SECONDS);
+                return promise;
+            }
+
+            @Override
+            public Future<Void> destroy(final Channel channel, final Promise<Void> promise) {
+                promise.trySuccess(null);
+                return promise;
+            }
+        };
+
+        final ApnsChannelPool pool = new ApnsChannelPool(factory, 1, EVENT_EXECUTOR, this.metricsListener);
+
+        pool.acquire();
+        pool.close().await();
+        assertTrue(createSuccess.isSuccess());
```

Comment:
I'm a little fuzzy on what this assertion is testing; can you explain in a little more detail?

Reply:
To assert that when the `ApnsChannelPool` close future is done, there's no create channel future waiting for fulfill.